engines/scumm/scumm.cpp

```diff
@@ -3479,10 +3480,15 @@ bool ScummEngine::displayMessageYesNo(const char *message, ...) {
 int ScummEngine_v90he::networkSessionDialog() {
 	GUI::MessageDialog dialog(_("Would you like to host or join a network play session?"), _("Host"), _("Join"));
 	int res = runDialog(dialog);
-	if (res == GUI::kMessageOK)
+	if (res == GUI::kMessageOK) {
 		// Hosting session.
-		return -1;
-
+		CreateSessionDialog createDialog;
+		if (runDialog(createDialog)) {
+			return -1;
+		} else {
+			return -2;
```

Comment:
As I understand, these magic numbers are processed by the SCUMM scripts. Still, of possible, I'd prefer turning them into enums.